### DISCUSSION TITLE
parse a sparql json result set to a TupleQueryResult

### DISCUSSION THREAD

**Discussion by D063520** [eclipse-rdf4j/rdf4j#3670] [Q&A] [answered]
Hi, I'm struggling in parsing a sparql json result string to a TupleQueryResult. Can someone help me out?

**Comment by barthanssens** [comment]
Sure, is there some part of the code you'd be willing to share ?

FWIW, depending on what you want to achieve, it could be worthwhile to look into some of the Repositories convenience methods (e.g just getting the results as a list: https://rdf4j.org/documentation/programming/repository/#a-tuple-query-in-a-single-line-of-code-the-repositories-utility)

**Comment by D063520** [comment]
Hi : ),

sorry ... there is not so much code to share. I have a file containing something like this:

{
  "head" : {
    "vars" : [
      "prg",
      "cci",
      "totp",
      "totbdg"
    ]
  },
  "results" : {
    "bindings" : [
      {
        "totp" : {
          "datatype" : "http://www.w3.org/2001/XMLSchema#integer",
          "type" : "literal",
          "value" : "33286"
        },
        "prg" : {
          "type" : "uri",
          "value" : "https://linkedopendata.eu/entity/Q2463066"
        },
        "cci" : {
          "type" : "literal",
          "value" : "2014BG16RFOP002"
        },
        "totbdg" : {
          "datatype" : "http://www.w3.org/2001/XMLSchema#decimal",
          "type" : "literal",
          "value" : "2340209884.15"
        }
      },
      {
        "totp" : {
          "datatype" : "http://www.w3.org/2001/XMLSchema#integer",
          "type" : "literal",
          "value" : "66"
        },
        "prg" : {
          "type" : "uri",
          "value" : "https://linkedopendata.eu/entity/Q2463063"
        },
        "cci" : {
          "type" : "literal",
          "value" : "2014BG16M1OP001"
        },
        "totbdg" : {
          "datatype" : "http://www.w3.org/2001/XMLSchema#decimal",
          "type" : "literal",
          "value" : "1997197541.69"
        }
      },


and I want to get out of it a TupleQueryResult?

**Reply by hmottestad** [reply]
How did you produce this file?

**Reply by D063520** [reply]
it is the result of a sparql query that I stored to disk ....

**Reply by hmottestad** [reply]
I think the easiest solution is to rerun the query against your SPARQL endpoint through an RDF4J SPARQLRepository: https://rdf4j.org/documentation/programming/repository/#sparql-endpoints

**Reply by hmottestad** [reply]
Some digging into the code turned up `SPARQLResultsJSONParserFactory`. But this is deep into the internals of RDF4J and not something I would recommend you consider using unless you are comfortable with figuring out a bunch of mostly undocumented code.

**Comment by D063520** [comment, marked as answer]
@hmottestad thank you for the pointer! I managed with

SPARQLResultsJSONParser sparqlResultsJSONParser = new SPARQLResultsJSONParser();
TupleQueryResultBuilder tupleQueryResultHandler = new TupleQueryResultBuilder();
sparqlResultsJSONParser.setQueryResultHandler(tupleQueryResultHandler);
sparqlResultsJSONParser.parseQueryResult(**inputstream**);
TupleQueryResult tupleQueryResult = tupleQueryResultHandler.getQueryResult();
